core/sdk/src/api/context.rs

```diff
@@ -76,27 +78,71 @@ impl RippleContext {
         RippleContext::get_from_payload(msg.clone())
     }
 
-    pub fn update(&mut self, request: RippleContextUpdateRequest) {
+    pub fn update(&mut self, request: RippleContextUpdateRequest) -> bool {
         match request {
             RippleContextUpdateRequest::Activation(a) => {
-                self.activation_status = a.into();
-                self.update_type = Some(RippleContextUpdateType::ActivationStatusChanged);
+                let activation_status: ActivationStatus = a.into();
```

Comment:
Nitpick but you could probably simplify these, e.g. something like:

`
        let activation_status: ActivationStatus = a.into();
        if let Some(status) = self.activation_status {
            if status == &activation_status {
                return false
            }
        }
        self.activation_status = Some(activation_status);
        self.update_type = Some(RippleContextUpdateType::ActivationStatusChanged);
        true
`

...maybe similar for some of the other below as well.